contracts/Depository.sol

```diff
@@ -226,56 +234,55 @@ contract Depository is IErrorsTokenomics {
 
         // Push newly created bond product into the list of products
         productId = productCounter;
-        mapBondProducts[productId] = Product(uint160(priceLP), uint32(expiry), token, uint96(supply));
+        mapBondProducts[productId] = Product(uint160(priceLP), uint32(vesting), token, uint96(supply));
         // Even if we create a bond product every second, 2^32 - 1 is enough for the next 136 years
         productCounter = uint32(productId + 1);
-        emit CreateProduct(token, productId, supply, priceLP, expiry);
+        emit CreateProduct(token, productId, supply, priceLP, vesting, block.timestamp);
     }
 
     /// @dev Closes bonding products.
     /// @notice This will terminate programs regardless of their vesting time.
     /// @param productIds Set of product Ids.
+    /// @return numClosedProducts Number of closed products.
     /// #if_succeeds {:msg "productCounter not touched"} productCounter == old(productCounter);
-    /// #if_succeeds {:msg "success closed"} forall (uint k in productIds) mapBondProducts[productIds[k]].expiry == 0 && mapBondProducts[productIds[k]].supply == 0;
-    function close(uint256[] memory productIds) external {
+    /// #if_succeeds {:msg "success closed"} forall (uint k in productIds) mapBondProducts[productIds[k]].vesting == 0 && mapBondProducts[productIds[k]].supply == 0;
+    function close(uint256[] memory productIds) external returns (uint256 numClosedProducts) {
         // Check for the contract ownership
         if (msg.sender != owner) {
             revert OwnerOnly(msg.sender, owner);
         }
 
         for (uint256 i = 0; i < productIds.length; ++i) {
             uint256 productId = productIds[i];
-            // Check if the product is still open
-            if (mapBondProducts[productId].expiry == 0) {
-                revert ProductClosed(productId);
-            }
-
+            // Check if the product is still open by getting its supply amount
             uint256 supply = mapBondProducts[productId].supply;
-            // Refund unused OLAS supply from the program if it was not used by the product completely
+            // The supply is greater than zero only if the product is active, otherwise it is instantly closed
             if (supply > 0) {
+                // Refund unused OLAS supply from the product if it was not used by the product completely
                 ITokenomics(tokenomics).refundFromBondProgram(supply);
-            }
-            address token = mapBondProducts[productId].token;
-            delete mapBondProducts[productId];
+                address token = mapBondProducts[productId].token;
+                delete mapBondProducts[productId];
 
-            emit CloseProduct(token, productId);
+                numClosedProducts++;
+                emit CloseProduct(token, productId, block.timestamp);
+            }
```

Comment:
Removed revert for trying to close an already-closed product. Also reduced the check for the supply. If the supply is zero, the product is already closed, otherwise the remaining amount is sent back to tokenomics.